### PR TITLE
Separate bemtree code of issue and comment blocks by their elem and mod files

### PR DESCRIPTION
**Pull request by tormozz48**
@tavriaforever

### CODE REVIEW

File: desktop.blocks/issue/__owner-action/issue__owner-action.bemtree [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+   match(function() { return this._isOwner; }).content()(function() {
+       return [
+           {
+               elem: 'edit'
+           },
+           {
+               elem: 'remove'
+           }
+       ]
```
Comment (tadatuta) [inline desktop.blocks/issue/__owner-action/issue__owner-action.bemtree]: ;

File: desktop.blocks/issue/__read-only/issue__read-only.bemtree [hunk 2]
```diff
@@ -0,0 +1,9 @@
+elem('read-only')(
+    match(function() { return !this._isArchive }).def()(function() {
```
Comment (tadatuta) [inline desktop.blocks/issue/__read-only/issue__read-only.bemtree:2]: can't we do this check at the parent level and do not create this elem if it's not archive?

File: desktop.blocks/issue/__owner-action/issue__owner-action.bemtree [hunk 3]
```diff
@@ -0,0 +1,16 @@
+block('issue').elem('owner-action')(
+   match(function() { return !this._isOwner; }).def()(function() {
```
Comment (tadatuta) [inline desktop.blocks/issue/__owner-action/issue__owner-action.bemtree:2]: can't we do this check at the parent level and do not create this elem if it's not an owner?

File: desktop.blocks/issue/__edit-form/issue__edit-form.bemtree [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                            text: 'Отменить',
+                            mix: [
+                                { block: 'issue', elem: 'edit-cancel' },
+                                { block: 'forum-form', elem: 'cancel' }
+                            ]
+                        }
+                    ]
+                }
+            ]
+        }
```
Comment (tadatuta) [inline desktop.blocks/issue/__edit-form/issue__edit-form.bemtree]: ;

**Comment by tadatuta** [comment]
it would be great to avoid `def()` matches if it's possible

File: desktop.blocks/issue/__read-only/issue__read-only.bemtree (hunk 2 above)
Comment (tormozz48) [inline desktop.blocks/issue/__read-only/issue__read-only.bemtree:2]: I want to disagree with you. As for me it's right way when every entity can incapsulate logic for displaying itself.

Comment (tadatuta) [inline desktop.blocks/issue/__read-only/issue__read-only.bemtree:2]: that's true when there is an entity. but in our case there's no reason for this entity to appear at all.

File: desktop.blocks/issue/__comments/issue__comments.bemtree [hunk 5]
```diff
@@ -0,0 +1,38 @@
+block('issue').elem('comments')(
+   match(function() { return this._forum && this._forum.comments; }).def()(function() {
+       var forum = this._forum,
+           issue = forum.issue,
+           comments = forum.comments;
+
```
Comment (tavriaforever) [inline desktop.blocks/issue/__comments/issue__comments.bemtree:6]: extra string

File: desktop.blocks/issue/__comments/issue__comments.bemtree [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+       return applyCtx([
```
Comment (tavriaforever) [inline desktop.blocks/issue/__comments/issue__comments.bemtree:8]: we can write it without array

File: desktop.blocks/comment/__avatar/comment__avatar.bemtree [hunk 7]
```diff
@@ -0,0 +1,12 @@
+block('comment').elem('avatar')(
+    def()(function() {
+        return applyCtx({
+            block: 'forum-user',
+            elem: 'avatar',
+            url: this.ctx.url,
+            alt: this.ctx.alt,
+            link: this.ctx.link,
+            mix: [{ block: 'comment', elem: 'avatar' }]
```
Comment (tavriaforever) [inline desktop.blocks/comment/__avatar/comment__avatar.bemtree:9]: mix value can be object - `mix: { block: 'comment', elem: 'avatar' }`

File: desktop.blocks/comment/__avatar/comment__avatar.bemtree [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+        });
+    })
+)
```
Comment (tavriaforever) [inline desktop.blocks/comment/__avatar/comment__avatar.bemtree:12]: need a semicolon

File: desktop.blocks/comment/__body/coment__body.bemtree [hunk 9]
```diff
@@ -0,0 +1,6 @@
+block('comment').elem('body')(
+    content()(function() {
+        var text = this.ctx.body;
+        return req.util.mdToHtml(text);
+    })
+)
```
Comment (tavriaforever) [inline desktop.blocks/comment/__body/coment__body.bemtree:6]: need a semicolon

File: desktop.blocks/comment/__date/comment__date.bemtree [hunk 10]
```diff
@@ -0,0 +1,9 @@
+block('comment').elem('date')(
+    def()(function() {
+        return applyCtx({
+            block: 'forum-date',
+            date: this.ctx.date,
+            mix: [{ block: 'comment', elem: 'date' }]
+        });
+    })
+)
```
Comment (tavriaforever) [inline desktop.blocks/comment/__date/comment__date.bemtree:9]: need a semicolon

File: desktop.blocks/comment/__date/comment__date.bemtree [hunk 11]
```diff
@@ -0,0 +1,9 @@
+block('comment').elem('date')(
+    def()(function() {
+        return applyCtx({
+            block: 'forum-date',
+            date: this.ctx.date,
+            mix: [{ block: 'comment', elem: 'date' }]
```
Comment (tavriaforever) [inline desktop.blocks/comment/__date/comment__date.bemtree:6]: mix value can be object

File: desktop.blocks/comment/__edit-form/comment__edit-form.bemtree [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+                            mods: { theme: 'normal', size : 'm' },
+                            text: 'Отменить',
+                            mix: [{ block: 'comment', elem: 'edit-cancel' }]
+                        }
+                    ]
+                }
+            ]
+        };
+    })
+)
```
Comment (tavriaforever) [inline desktop.blocks/comment/__edit-form/comment__edit-form.bemtree:65]: need a semicolon

File: desktop.blocks/comment/__header/comment__header.bemtree [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+            {
+                elem: 'owner-actions',
+                content: [
+                    { elem: 'edit' },
+                    { elem: 'remove' }
+                ]
+            }
+        ];
+    })
+)
```
Comment (tavriaforever) [inline desktop.blocks/comment/__header/comment__header.bemtree:30]: need a semicolon

File: desktop.blocks/comment/__username/comment__username.bemtree [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+    def()(function() {
+        return {
+            block: 'forum-user',
+            elem: 'name',
+            url: this.ctx.url,
+            mix: [{ block: 'comment', elem: 'username' }],
+            content: this.ctx.name
+        };
+    })
+)
```
Comment (tavriaforever) [inline desktop.blocks/comment/__username/comment__username.bemtree:11]: need a semicolon

File: desktop.blocks/issue/__comments/issue__comments.bemtree [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+               issueNumber: issue.number,
+               comments: issue.comments,
+               forumUrl: req.forumUrl
+           },
+           content: {
+               elem: 'container'
+           }
+       });
+   })
+)
```
Comment (tavriaforever) [inline desktop.blocks/issue/__comments/issue__comments.bemtree:38]: need a semicolon

File: desktop.blocks/issue/__labels/issue__labels.bemtree [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+
+        return labels.map(function(label) {
+            return {
+                elem: 'label',
+                color: label.color,
+                content: label.name
+            }
+        });
+    })
+)
```
Comment (tavriaforever) [inline desktop.blocks/issue/__labels/issue__labels.bemtree:14]: need a semicolon

File: desktop.blocks/issue/__owner-action/issue__owner-action.bemtree [hunk 17]
```diff
@@ -0,0 +1,12 @@
...
+       return '';
+   }),
+
+   match(function() { return this._isOwner; }).content()(function() {
+       return [
+           { elem: 'edit' },
+           { elem: 'remove' }
+       ];
+   })
+)
```
Comment (tavriaforever) [inline desktop.blocks/issue/__owner-action/issue__owner-action.bemtree:12]: need a semicolon

**Comment by tavriaforever** [comment]
looks good, I`m waiting for merge this PR

**Pull request closed without merging**